### DISCUSSION TITLE
Pre-Check rating_config.json while delivering to AppStore

### DISCUSSION THREAD

**Discussion by patreu22** [fastlane/fastlane#17386] [Ideas] [answered]
Hi there,  

I was switching from the deprecated rating_config.json to the new one, an example is provided in the repo itself:
https://github.com/fastlane/fastlane/blob/master/deliver/assets/example_rating_config.json

While knowing, that it's totally my fault, I basically copy/pasted this rating without paying any further attention to it. I updated a couple of apps and the next day I realized, that the apps went lost on the AppStore. The reason is, that with the example rating_config, the app will pass the AppStore test, but eventually not getting listed. It's not even reachable through the link provided by the AppStore.

My suggestion is to add a warning to the metadata pre-check feature of deliver that checks for the age-rating that is present in AppStoreConnect. In case the developer provides values that might delist the App from the AppStore, he should receive a warning to be aware of the situation. As I said, that one was on me, but such a feature would've helped a lot.

**Comment by rogerluan** [comment, marked as answer]
Hey @patreu22 👋 

Your suggestion sounds valid, I'd be in favour of it. However, it's not clear to me why your app got "lost" in the App Store? How the contents of that json file made your app be unlisted? Is it because it gave the wrong age rating and Apple banned it?

**Reply by patreu22** [reply]
Hey Roger 👋

>Is it because it gave the wrong age rating and Apple banned it?

I think that's it. If you check the example I think that's critical:
```
"violenceRealisticProlongedGraphicOrSadistic": "FREQUENT_OR_INTENSE"
```

To be honest I didn't do any research so far regarding Apples policies and when they ban an app. What confused me was that they approved it, but didn't display it on the AppStore. The only hint was a small badge in the age rating menu.  

For a first approach I would suggest to display a little warning in the case that any value is not `NONE` or `false`.

**Reply by rogerluan** [reply]
@patreu22 I didn't notice that specific setting, on a first glance I thought the template suggested a "Age 4+" rating (i.e. everything set to false/minimum). 

Isn't your Apple ID configured to not display violent apps? Or, is it underage? 

Can you share your app name or link with me so I can search for it? 

There are valid cases where users will want to use those settings (in case their app actually displays violence, or such content), so that check might not be the best solution. Probably just changing the default/template json would already help a ton 😄

**Reply by patreu22** [reply]
Unfortunately I can't share the app title with you, but I got it confirmed by 3 other people that they were not able to find it. Our fixing updates already passed and the apps start to re-appear (at least they show up while clicking on the AppStore link, indexing the keywords didn't happen yet). I just use a regular Apple ID without any security features.

The only screenshot I got left is this one here (sorry for the German):
![image](https://user-images.githubusercontent.com/11362357/95216184-11614700-07f2-11eb-9c41-66d511703b7b.png)

It basically says "Because of the chosen age rating this app won't be offered in the AppStore". With all the other indications it was an issue for the global store and not related to my Apple ID. For sure, in a first step changing the json would be great, but I think at least a warning (doesn't have to abort the lane) would be helpful to make developers aware their app might be ghosted by Apple.

**Reply by rogerluan** [reply]
That cleared things up! I found the answer in this tech crunch article: https://techcrunch.com/2009/06/29/heres-how-iphone-app-store-ratings-work-hint-they-dont/

Explains a lot, although it's really weird and behaves oddly.

Now I'm certain that we can add a check for this because, if any of those 2 show anything other than NONE, your app **will not** be  offered in the App Store. So, yeah, worth a check (with an error, preventing you from proceeding).

I'll create a PR for this 😊 thanks for guiding me through this, @patreu22 !

**Reply by patreu22** [reply]
Cheers, thanks for taking over from here!
